### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by shuyuduan**
Student: Shuyu Duan, s4552709， problem：Segment the OASIS brain data set with an Improved UNet [1] with all labels having a minimum Dice
similarity coefficient of 0.9 on the test set

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Most commits made made after deadline
- Code seems reasonable for Easy Difficult
- UNRUNNABLE code main.py
- Results seem good
- README.MD in the wrong directory: you should place it in your own directory
- NO example of results, unable to verify validity.
*Other Feedback*: 
- Overly generic commit messages

File: recognition/s4552709/predict.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+def model_prediction(x_test):
```
Comment (shakes76) [inline recognition/s4552709/predict.py:1]: No imports

File: recognition/s4552709/model.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
+import tensorflow as tf
```
Comment (shakes76) [inline recognition/s4552709/model.py:1]: Code header block should be added for all scripts/source files

File: recognition/s4552709/dice.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+def dice_coefficient(y_true, y_pred, smooth=0.0001):
+    #change the dimension to tree
+    y_true_f =tf.keras.backend.flatten(y_true)
+    y_pred_f = tf.keras.backend.flatten(y_pred)
+    #calculation for the loss function
+    intersection = tf.keras.backend.sum(y_true_f * y_pred_f)
+    return (2. * intersection + smooth) / (tf.keras.backend.sum(y_true_f) + tf.keras.backend.sum(y_pred_f) + smooth)
+
+def dice_coefficient_loss(y_true, y_pred):
+      return 1. - dice_coefficient(y_true, y_pred)
```
Comment (shakes76) [inline recognition/s4552709/dice.py:12]: Indent unusual

**Comment by shakes76** [comment]
## TF Usage
Not good
## Good Practice (Design/Commenting)
OK design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme sparse
Module present
## Commit Log
Meaningful commit messages 
Progressive not commits used 
## Documentation
Description and Comments OK
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. Fix imports and indentation where mentioned.**
Fixing the above will only result in obtaining additional feedback marks according to the assessment sheet.
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.
